api/src/Service/CampCouponService.php

```diff
@@ -2,23 +2,54 @@
 
 namespace App\Service;
 
+use App\Entity\Camp;
+use Doctrine\ORM\EntityManagerInterface;
+
 class CampCouponService {
-    public function __construct(private string $secret) {
+    public function __construct(
+        private string $secret,
+        private EntityManagerInterface $em
+    ) {
     }
 
     public function createCoupon() {
         if ('' == $this->secret) {
             return 'No Coupon-Key required';
         }
+        $min = ceil(78364164096 / intval($this->secret));
+        $max = floor(2821109907455 / intval($this->secret));
+        $rnd = mt_rand($min, $max) * intval($this->secret);
+        $coupon = base_convert($rnd, 10, 36);
```

Comment:
Nice algorithm. A little math:
You left a comment saying that we need a secret prime number between 100000 and 1000000. Wolfram alpha says there are [68906](https://www.wolframalpha.com/input?i=number+of+prime+numbers+between+100000+and+1000000) such prime numbers which we can use. That is 16 bits of entropy, which is weaker than a [2-letter password](https://www.wolframalpha.com/input?i=2-letter+password). And that is if this algorithm, which you obviously found on the internet, really holds what it promises and has no bugs.

Wouldn't it be easier to just store a list of valid coupon keys either in a table in the DB or in a secret (or secret file)? Almost just as easy to generate new tokens, and way more secure. I know it's really not that critical in this case. But I would never recommend writing security-relevant algorithms yourself, and if we keep this in the code, I really hope it is never ever reused for some other security-relevant calculations... E.g. in case we ever add more tokens in the future for other purposes for some reason.

Reply:
## Core Team Decision
Add comment: don't copy this, PLEASE :D